File: Samples/Scripts/Spark/Python/GettingStarted/JSON.py
from pyspark import SparkContext
from pyspark.sql import SQLContext

sqlContext = SQLContext(sc)

#A JSON dataset is pointed to by path.
#The path can be either a single text file or a directory storing text files.

path = "/Data/Spark/Resources/People.json"
people = sqlContext.read.json(path)

#The inferred schema can be visualized using the printSchema() method.
people.printSchema()

#root  (Structure of people)
# |-- age: integer (nullable = true)
# |-- name: string (nullable = true)

#Register this DataFrame as a table.
people.registerTempTable("people")

teenagers = sqlContext.sql("SELECT name FROM people WHERE age >= 13 AND age <= 19")
teenNames = teenagers.rdd.map(lambda p: "Name: " + p.name)
for teenName in teenNames.collect():
	print (teenName)

anotherPeopleRDD = sc.parallelize(['{"name":"Yin","address":{"city":"Columbus","state":"Ohio"}}'])
anotherPeople = sqlContext.read.json(anotherPeopleRDD)

File: Samples/Scripts/Spark/Scala/GettingStarted/JSON.scala
// A JSON dataset is pointed to by path.
// The path can be either a single text file or a directory storing text files.
val path = "/Data/Spark/Resources/People.json"
val people = spark.read.json(path)

// The inferred schema can be visualized using the printSchema() method.
people.printSchema()
// root
//  |-- age: integer (nullable = true)
//  |-- name: string (nullable = true)

// Register this DataFrame as a table.
people.registerTempTable("people")

// SQL statements can be run by using the sql methods provided by sqlContext.
val teenagers = spark.sql("SELECT name FROM people WHERE age >= 13 AND age <= 19")

// Alternatively, a DataFrame can be created for a JSON dataset represented by
// an RDD[String] storing one JSON object per string.
val anotherPeopleRDD = sc.parallelize(
  """{"name":"Yin","address":{"city":"Columbus","state":"Ohio"}}""" :: Nil)
val anotherPeople = spark.read.json(anotherPeopleRDD)